lib/typing/decimal/details.go

```diff
@@ -0,0 +1,64 @@
+package decimal
+
+import (
+	"fmt"
+
+	"github.com/artie-labs/transfer/lib/ptr"
+)
+
+type DecimalDetails struct {
+	scale     int
+	precision *int
+}
+
+func NewDecimalDetails(precision *int, scale int) *DecimalDetails {
```

Comment:
This is exactly the same as `NewDecimal()` except that it doesn't take a third `value` argument.